src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1414,6 +1417,25 @@ def aks_browse(cmd, client, resource_group_name, name, disable_browser=False,
         dashboard_pod = str(dashboard_pod).split('/')[-1].strip()
     else:
         raise CLIError("Couldn't find the Kubernetes dashboard pod.")
+
+    # Find the port
+    try:
+        dashboard_port = subprocess.check_output(
+            ["kubectl", "get", "pods", "--kubeconfig", browse_path, "--namespace", "kube-system",
+             "--selector", "k8s-app=kubernetes-dashboard",
+             "--output", "jsonpath='{.items[0].spec.containers[0].ports[0].containerPort}'"]
+        )
+    except subprocess.CalledProcessError as err:
+        raise CLIError('Could not find dashboard port: {}'.format(err))
+    if dashboard_port:
+        # output format: b"'{port}'"
+        dashboard_port = int((dashboard_port.decode('utf-8').replace("'", "")))
```

Comment:
how do casts fail in python? presumable if it isn't graceful it will be caught by your outer try/expect. Can you check/confirm?